new-client/src/components/App.js

```diff
@@ -1038,11 +1038,23 @@ class App extends React.PureComponent {
     );
   };
 
-  checkDrawerButtons() {
-    return (
-      !this.state.drawerStatic ||
-      (this.state.drawerStatic && this.state.drawerButtons.length > 1)
+  showDrawerButtons() {
+    const drawerButtons = this.state.drawerButtons;
+
+    if (!drawerButtons) return false;
+
+    // We check if the plugin button (or any button) is empty and then subsequently hidden
+    const isHiddenPluginPresent = drawerButtons.some(
+      (button) => button.hideOnMdScreensAndAbove
     );
+
+    // We want to check if there's only one visible drawerButton
+    const isOnlyOneButtonVisible =
+      drawerButtons.length === 1 ||
+      (drawerButtons.length === 2 && isHiddenPluginPresent);
```

Comment:
Hi! This slipped my attention prior the merge, but anyway: how about a situation which more than 2 drawer buttons? I know that we currently only have two of them, but since any plugin can add such a button (check out the Dummy plugin…) it's not impossible to end up with 3 or more buttons in the future. Have you checked for compatibility in such a situation? Thanks! /J

Reply:
Hello! I think in a situation where we have 2 or more visible buttons (and if you have 3 buttons they probably are all visible) the buttons should not be removed if the drawer is static. This logic only ensures that if there is one button, it doesn't show, OR if there are two buttons (one visible and one invisible) they dont show.

Reply:
But might be an idea to change logic anyways in the event of something like that :)